src/coreclr/jit/importer.cpp

```diff
@@ -9836,6 +9836,10 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                     if (op2->IsIntegralConst())
                     {
                         const ssize_t allocSize = op2->AsIntCon()->IconValue();
+                        if ((unsigned)allocSize > UINT_MAX)
+                        {
+                            BADCODE("Localloc size is negative or too big");
```

Comment:
I don't think this `BADCODE` is spec-compliant, since the size is both an unsigned integer and can be `ulong` on 64 bit platforms.

So it's more of an `IMPL_LIMITATION` in that sense.

But even with that, we should not crash the compiler based on facts influenced by optimizations.

I think we can avoid this by using `STORE_DYN_BLK` if the size is too large?

Reply:
Good point, I think I'll just not mark size as Contained so it will go through the "variable" size path everywhere else